### ISSUE TITLE
Skill Variable Issue

### ISSUE THREAD

**Issue by Sunaries** [Kagamia/WzComparerR2#38]
Kanna in GMS will soon receive a buff. I compared the old and new version and I noticed that some numbers are missing in the new version.

Example)

**BEFORE ↓**
![1](https://cdn.discordapp.com/attachments/236180502200057856/345992574072520704/before3.png)
_Skill description_

![2](https://cdn.discordapp.com/attachments/236180502200057856/345992576480051202/before3_stat.png)
_Variables_

**AFTER ↓**
![3](https://cdn.discordapp.com/attachments/236180502200057856/345992568036786198/after3.png)

![4](https://cdn.discordapp.com/attachments/236180502200057856/345992569521700874/after3_stat.png)

If you read the skill description in the BEFORE version, you will read: `Haku's Blessing - DEF: +200, Hakku's Fan ATT: +20%, Duration: 40 Sec.`
However, in the updated version, you will read: `Haku's Blessing - DEF: , Hakku's Fan ATT: +20%, Duration: 40 Sec.` - The value after DEF is missing.

This happens to other values as well, take a look at the next example.

**BEFORE ↓**
![1](https://cdn.discordapp.com/attachments/236180502200057856/345993920699695104/before1.png)
_Skill description_

![2](https://cdn.discordapp.com/attachments/236180502200057856/345993924201938946/before1_stat.png)
_Variables_

**AFTER ↓**
![3](https://cdn.discordapp.com/attachments/236180502200057856/345993917742710785/after1.png)

![4](https://cdn.discordapp.com/attachments/236180502200057856/345993919399591936/after1_stat.png)

**Comment by KENNYSOFT** [comment]
Do you re-initialize StringLinker with new version? Variable names linked to skill description is stored in string.wz, so you should do that.

**Comment by Sunaries** [comment]
@KENNYSOFT I actually opened two WCR2. One loading the current version and one with the new update.

**Comment by Kagamia** [comment]
(That's the reason why I leave a default '0' there 😆  )
[SummaryParser.cs](https://github.com/Kagamia/WzComparerR2/blob/master/WzComparerR2.Common/CharaSim/SummaryParser.cs#L118)

**Comment by Sunaries** [comment]
@Kagamia Ah, @KENNYSOFT commented that line. If I uncomment it, I will see this:

![test](https://cdn.discordapp.com/attachments/236180502200057856/352222919213973505/333.png)

`indieIgnoreMobpdpR` 25 + x (x=15) = 40% - Yet it outputs as 0%.

**Comment by Sunaries** [comment]
@KENNYSOFT said he commented `sb.Append(0);//默认值` due to the following issue:

![2](https://cdn.discordapp.com/attachments/88992151643369472/352600810225532949/skill_400001024_comment.png)
↑ comment

![1](https://cdn.discordapp.com/attachments/88992151643369472/352600827455733771/skill_400001024_uncomment.png)
↑ uncomment

String: 
```
MP Cost: #mpCon, Duration: #s sec, Max Stacks: #x, Each stack provides an additional buff.\n#cStack 1# - Skill Cooldown: -#w% (excludes skills unaffected by cooldown reduction)\n#cStack 2# -  Knockback Resistance: +#z%\n#cStack 3# - All Stats: +#q\n#Stack 4# - Attack Power &amp; Magic ATT: +#u\n#Stack 5# - Boss Damage: +#v%\n#Stack 6#: Invincible for duration of skill\nCooldown: #cooltime sec. Cooldown increased to #y sec when buff expires (including character death) or max stack count is reached."
```

It should say `Stack 4`, yet it doesn't on both outputs.

**Comment by Kagamia** [comment]
@Sunaries 
 I've simply tested with the skill summary. 
Yes, this annoying bugs maybe only appear from GMS/MSEA client. 😞 

Here explains the working model of [SummaryParser](https://github.com/Kagamia/WzComparerR2/blob/master/WzComparerR2.Common/CharaSim/SummaryParser.cs) class:
1> Find a '#' character. (L26)
2> Get a word after '#' as the property name, as long as it could be, allows empty. (L29-42)
3> Matching with `common` properties. (L45-56)
4> If failed, matching with global reserved properties. (L64-89)
5> If failed, and already in orange font color (beginC), then close it. (L91-96)
6> Else, a 'c' character is after '#', then begin orange color. (L97-102)
7> If not find a word after '#' **(important!!!!!!!!! A single '#' character is treated as '#c')**, the same as <6>. (L103-108)
8> If all failed, consider that the given property name is missing, return a '0' as default. (L109-121)

The issue is, <7> will never take effect, because in GMS/MSEA, It could always find an valid property name after '#' at most time (they are always English characters 😃), if I fix that, <8> could never be triggered.


@KENNYSOFT 
I need both common properties and skill H summary in string.wz for testing.

**Comment by KENNYSOFT** [comment]
Right. So we should make assumption like 'all of property names are like camelCase'; which starts with small letters.

**Comment by Kagamia** [comment]
So, what is the right order? 
````
 beginC(#c) -> single#(# as #c) -> endC(#) -> missing Property(#blabla = default(0))
````
Does it make sense?

I have to check all common property names in the skill.wz, Nexon make a joke again. 😆
